CRM/Contribute/BAO/ContributionSoft.php

```diff
@@ -410,91 +408,64 @@ public static function getSoftContributionList($contact_id, $filter = NULL, $isT
         'title' => ts('View related contribution'),
       ],
     ];
-    $orderBy = 'cc.receive_date DESC';
-    if (!empty($dTParams['sort'])) {
-      $orderBy = $dTParams['sort'];
+
+    $contributionSofts = ContributionSoft::get()
+      ->addSelect('*', 'contribution_id.receive_date', 'contribution_id.contact_id', 'contribution_id.contact_id.display_name', 'soft_credit_type_id:label', 'contribution_id.contribution_status_id:label', 'contribution_id.financial_type_id:label', 'pcp_id.title')
+      ->addWhere('contact_id', '=', $contact_id)
+      ->addWhere('contribution_id.is_test', '=', $isTest);
+
+    if ($membershipID) {
+      $contributionSofts->addJoin('LineItem', 'INNER', NULL,
```

Comment:
@eileenmcnaughton I know that this is the right direction but I'm just wondering if we need to do a status check / upgrade to check that every membership - contribution linkage in civicrm_membership_payment has a correct line item record?

Reply:
@seamuslee001 maybe but not specifically for this PR I don't think - there are other places in the code that expect valid data in lineitems